lib/compress.js

```diff
@@ -2436,31 +2461,29 @@ merge(Compressor.prototype, {
             }
             break;
         }
-        if (compressor.option("conditionals")) {
-            if (self.operator == "&&") {
-                var ll = self.left.evaluate(compressor);
-                if (ll.length > 1) {
-                    if (ll[1]) {
-                        compressor.warn("Condition left of && always true [{file}:{line},{col}]", self.start);
-                        return maintain_this_binding(compressor.parent(), self, self.right.evaluate(compressor)[0]);
-                    } else {
-                        compressor.warn("Condition left of && always false [{file}:{line},{col}]", self.start);
-                        return maintain_this_binding(compressor.parent(), self, ll[0]);
-                    }
+        if (compressor.option("conditionals")) switch (self.operator) {
```

Comment:
I originally wrote that condition `if (compressor.option("conditionals"))` and it's incorrect. I later learned that the term "conditional" in a javascript context means "ternary operator" which has nothing to do with short-circuiting conditions. The "booleans" option is not appropriate either because it deals specifically with the `true`/`false` -> `!0`/`!1` transform. I suggest to simply remove this `if` altogether. Might impact some tests though.